_rules/image-accessible-name-23a2a8.md

```diff
@@ -72,106 +72,118 @@ _There are currently no assumptions._
 
 ```html
-<img title="W3C logo" />
+<img title="W3C logo" src="/test-assets/shared/w3c-logo.png" />
 ```
 
 #### Passed Example 5
 
-The HTML `img` element has a [semantic role][] of `presentation` through an empty `alt` attribute.
+This `img` element has an [implicit role][] of `presentation` because of the empty `alt` attribute.
 
 ```html
-<img alt="" />
+<img alt="" src="/test-assets/shared/background.png" />
 ```
 
 #### Passed Example 6
 
-The HTML `img` element has a [semantic role][] of `presentation` through [explicit role][].
+The `img` element has an [explicit role][] of `presentation` because of the value of the `role` attribute.
 
 ```html
-<img role="presentation" />
+<img role="presentation" style="width:72px; height:48px; background-image: url(/test-assets/shared/background.png)" />
 ```
 
 #### Passed Example 7
 
-The HTML `img` element has a [semantic role][] of `none` through [explicit role][].
+The `img` element has an [explicit role][] of `none` because of the value of the `role` attribute.
 
 ```html
-<img role="none" />
+<img role="none" src="/test-assets/shared/background.png" />
 ```
 
 #### Passed Example 8
 
-The HTML `img` element has an [accessible name][] that is not empty.
+This `img` element inside a `div` positioned off screen has an [implicit role][] of `img` because of the empty `alt` attribute.
 
 ```html
-<img alt=":-)" />
+<div style="margin-left:-9999px;">
+	<img alt="" src="/test-assets/shared/background.png" />
+</div>
 ```
 
 ### Failed
 
 #### Failed Example 1
 
-The HTML `img` element with [implicit role][] of `img` has an empty [accessible name][].
+This `img` element has an empty [accessible name][] and an [implicit role][] of `img` because it is missing an empty `alt` attribute.
 
 ```html
-<img />
+<img src="/test-assets/shared/w3c-logo.png" />
 ```
 
 #### Failed Example 2
 
-The element with role of `img` has an empty [accessible name][].
+This element with role of `img` has an empty [accessible name][].
 
 ```html
-<div role="img"></div>
+<div role="img" style="width:72px; height:48px; background-image: url(/test-assets/shared/w3c-logo.png)"></div>
 ```
 
 #### Failed Example 3
 
-The `img` element inside a `div` positioned off screen has an empty [accessible name][] and an [implicit role][] of `img`.
+This `img` element inside a `div` positioned off screen has an empty [accessible name][] and an [implicit role][] of `img`.
 
 ```html
-<div style="margin-left:-9999px;"><img /></div>
+<div style="margin-left:-9999px;"><img src="/test-assets/shared/w3c-logo.png" /></div>
 ```
 
 #### Failed Example 4
 
-The HTML `img` element has an empty [accessible name][].
+This `img` element has an empty [accessible name][] because the space in the `alt` attribute is trimmed off by the [accessible name computation](https://www.w3.org/TR/accname-1.1/).
```

Comment:
Do we want to expand the description to say that even if it is trimmed by accname computation, it still prevent `alt` to be empty for role mappings?

Reply:
Yeah, good idea